packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -82,19 +82,17 @@ impl<'id> MemoryId<'id> {
     }
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
-    pub fn clean_up<Id: Borrow<Uuid>>(id: Id) -> Result<()> {
+    pub fn clean_up<Id: Borrow<Uuid>>(id: Id) {
         // TODO: macOS does not store the shared memory FDs at `/dev/shm/`. Maybe it's not storing
         //   FDs at all. Find out if they are stored somewhere and remove them instead, otherwise we
         //   have to figure out a way to remove them without relying on the file-system.
-        let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)))
-            .map_err(|e| Error::Unique(format!("cleanup glob error: {}", e)))?;
+        let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)));
 
-        shm_files.filter_map(Result::ok).for_each(|path| {
+        shm_files.into_iter().flatten().flatten().for_each(|path| {
```

Comment:
Are we now doing an `into_iter` over the result of the glob? That seems strange to me

Reply:
We are ignoring the error, relevant comments: https://github.com/hashintel/hash/pull/477#discussion_r848332451 https://github.com/hashintel/hash/pull/477#discussion_r848334516
This way if there is a glob error, the whole cleanup doesn't stop.

Reply:
I don't disagree with ignoring the error, but the code layout is not clear about what's going on here at all, and the `into_iter().flatten().flatten()` is a pretty big smell.

We should be explicitly ignoring the error, and adding a comment to say why.

Reply:
I agree that there are many layers of iterators and `Result` which makes this code complicated and a comment explaining the situation would be useful. But I don't agree it's a code smell, using multiple `flatten` is literally in [its doc](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.flatten).

For the types:
- `glob::glob` returns a `Result` that errors when the pattern is incorrect
- then you have `Paths`, an iterator over the path matching the pattern
- `Paths` yields `Result`s that error when the path can't be accessed

The first error shouldn't happen and would be a bug on our side. For the second one we could warn but I'm guessing these files are going to be removed by the OS anyway.

Reply:
Made the error ignoring a bit more obvious and added a comment in https://github.com/hashintel/hash/pull/494

Reply:
The `flatten` in the docs are used for some nested `Vec`s, not for different types. I know that I brought `flatten` into discussion but seeing the result feels hard to follow the code without type hints. Maybe we should just revert back to `filter` instead. It's a little bit more verbose but readable without guessing about the underlying type.